misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormatterUtils.java

```diff
@@ -15,25 +15,154 @@
+
+        if (Files.exists(cachePath)) {
+            try {
+                return Files.readString(cachePath, StandardCharsets.UTF_8);
+            } catch (IOException e) {
+                throw new FormatterException("Failed to read cached formatting configuration file");
+            }
+        }
+
+        try {
+            URL url = new URL(fileUrl);
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+
+            int responseCode = connection.getResponseCode();
+
+            if (responseCode == HttpURLConnection.HTTP_OK) {
+                try (BufferedReader reader = new BufferedReader(
+                        new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8))) {
+                    String line;
+                    while ((line = reader.readLine()) != null) {
+                        fileContent.append(line).append(NEWLINE_SYMBOL);
+                    }
+                }
+            } else {
+                throw new FormatterException("Failed to retrieve remote file. HTTP response code: " + responseCode);
+            }
+            connection.disconnect();
+        } catch (IOException e) {
+            throw new FormatterException("Failed to retrieve remote formatting configuration file");
+        }
+        cacheRemoteConfigurationFile(root, fileContent.toString());
+
+        return fileContent.toString();
+    }
+
+    private static void cacheRemoteConfigurationFile(Path root, String content) throws FormatterException {
+        if (Files.exists(root.resolve("target"))) {
+            if (!Files.exists(root.resolve("target").resolve("format"))) {
+                try {
+                    Files.createDirectories(root.resolve("target").resolve("format"));
+                } catch (IOException e) {
+                    throw new FormatterException("Failed to create format configuration cache directory");
+                }
+            }
+            String filePath = root.resolve("target").resolve("format").resolve("Format.toml")
+                    .toString();
+            try (BufferedWriter writer = new BufferedWriter(new FileWriter(filePath, StandardCharsets.UTF_8))) {
+                writer.write(content);
+            } catch (IOException e) {
+                throw new FormatterException("Failed to write format configuration cache file");
+            }
+        }
+    }
+
+    public static Map<String, Object> parseConfigurationToml(TomlDocument document) {
+        TomlTableNode tomlAstNode = document.toml().rootNode();
+        Map<String, Object> formatConfigs = new HashMap<>();
+        if (!tomlAstNode.entries().isEmpty()) {
+            Map<String, Object> tomlMap = document.toml().toMap();
+            for (Map.Entry<String, Object> entry : tomlMap.entrySet()) {
+                formatConfigs.put(entry.getKey(), entry.getValue());
+            }
+        }
+        return formatConfigs;
```

Comment:
```suggestion
        if (!tomlAstNode.entries().isEmpty()) {
            return document.toml().toMap();
        }
        return new HashMap<>();;
```
Do we need to return a copy of the toml map?